From 288b87f8f61ba44f4e499ae4ef9d6b053be0da90 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 23 Dec 2020 14:46:55 +0200
Subject: [PATCH] [SW-20154] goya2_arc: change the memcpy block's mask

Memcpy block's config mask has a different name in
Gaudi2, Instead of field size mask, it is renamed
to mem size mask. To accommodate this change, get
this from a function which is implemented specific
to each platform

Change-Id: I348d75c94de759eab9a4c617d023c088a2506681
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 979f30f..3210e2b 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -13,6 +13,7 @@
 #include "spi_flash.h"
 #include "pcie.h"
 #include "delay.h"
+#include <common.h>
 
 struct arcboot_global_s arcboot_rom_global = {
 	.config_fuse = 0,
@@ -68,6 +69,7 @@
 {
 	uint32_t mem_cfg, cpy_sts;
 	uint8_t retry = HL_INIT_SP_SRAM_RETRIES;
+	uint32_t cfg_mask = get_mem_cfg_mask();
 
 	/* Use memblock to initialize scratchpad memory */
 	regmap_writel((uint32_t)HL_SP_SRAM_BASE,
@@ -76,10 +78,9 @@
 			mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_H);
 
 	mem_cfg = regmap_readl(mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
-	mem_cfg &= ~(PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK);
+	mem_cfg &= ~(cfg_mask);
 	/* set size of 64k in granularity of 16 bytes */
-	mem_cfg |= (SP_SRAM_SIZE_16B_GRANULARITY &
-			PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK);
+	mem_cfg |= (SP_SRAM_SIZE_16B_GRANULARITY & cfg_mask);
 	mem_cfg |= (0x1 << PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_SHIFT);
 
 	regmap_writel(mem_cfg, mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
diff --git a/common/common.h b/common/common.h
new file mode 100644
index 0000000..1707161
--- /dev/null
+++ b/common/common.h
@@ -0,0 +1,11 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2020 HabanaLabs Ltd.
+ */
+
+#ifndef _COMMON_H_
+#define _COMMON_H_
+
+uint32_t get_mem_cfg_mask(void);
+
+#endif /* _COMMON_H_ */
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index fa3e927..cde0aa7 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -8,8 +8,14 @@
 #include <unistd.h>
 #include <efuse.h>
 #include <psoc_efuse_regs.h>
+#include <psoc_global_conf_masks.h>
 
 uint32_t get_efuse_read_reg(void)
 {
 	return mmPSOC_EFUSE_READ;
 }
+
+uint32_t get_mem_cfg_mask(void)
+{
+	return PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK;
+}
